server/src/controllers/article.controller.js

```diff
@@ -0,0 +1,111 @@
+import models from '../database/models';
+
+const { Articles } = models;
+/**
+   * create a new article
+   * @method createArticle
+   * @param {object} req
+   * @param {object} res
+   * @returns {(function|object)} Function next() or JSON object
+   */
+export const createArticle = async (req, res) => {
+  const { id: userId } = req.decoded;
+
+  const data = await Articles.create({ ...req.body, userId });
+
+  return res.status(201).json({ status: 'success', data });
+};
+
+
+/**
+   * create a get a single Article
+   * @method getSingleArticle
+   * @param {object} req
+   * @param {object} res
+   * @returns {(function|object)} Function next() or JSON object
+   */
+export const getSingleArticle = async (req, res) => {
+  const { id } = req.params;
+  const { id: userId } = req.decoded;
+
+  const foundArticle = await Articles.findOne({ where: { id, userId } });
+
+  if (!foundArticle) return res.status(404).json({ status: 'failure', error: 'article does not exist' });
+
+  return res.status(200).json({ status: 'success', data: foundArticle });
+};
+
+/**
+   * get all articles
+   * @method getAllArticles
+   * @param {object} req
+   * @param {object} res
+   * @returns {(function|object)} Function next() or JSON object
+   */
+export const getAllArticles = async (req, res) => {
+  const { id: userId } = req.decoded;
+  const { type } = req.query;
+
+
+  if (type === 'draft') {
+    const getAllDraft = await Articles.findAll({ where: { userId, type: 'draft' } });
+    return res.status(200).json({ status: 'success', getAllDraft });
+  }
+
+  if (type === 'published') {
+    const getAllPublished = await Articles.findAll({ where: { userId, type: 'published' } });
```

Comment:
Since the type coming from req.query is the same as the type in the db why not just use one block of code for this 

```suggestion
    const articles = await Articles.findAll({ where: { userId, type } });
```

Reply:
You'll check if the type field was passed first though